src/general/DayNightCycle.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using Newtonsoft.Json;
+
+[JsonObject(IsReference = true)]
+[UseThriveSerializer]
+public class DayNightCycle
+{
+    [JsonProperty]
+    public DayNightConfiguration LightCycleConfig;
+
+    [JsonProperty]
+    public bool IsEnabled;
+
+    /// <summary>
+    ///   The multiplier used for calculating DayLightPercentage.
+    /// </summary>
+    /// <remarks>
+    ///   This exists as it only needs to be calculated once and
+    ///   the calculation for it is confusing.
+    /// </remarks>
+    [JsonIgnore]
+    private float daytimeMultiplier;
+
+    public DayNightCycle(bool isEnabled)
+    {
+        IsEnabled = isEnabled;
+
+        LightCycleConfig = SimulationParameters.Instance.GetDayNightCycleConfiguration();
+        Time = LightCycleConfig.HoursPerDay / 2;
+
+        float halfPercentage = LightCycleConfig.DaytimePercentage / 2;
+
+        // This converts the percentage in DaytimePercentage to the power of two needed for DayLightPercentage
+        daytimeMultiplier = (float)Math.Pow(2, 1 / halfPercentage);
```

Comment:
`Mathf.pow`. Also, a source motivating this formula would be neat here.

Reply:
Being honest, I don't actually know why this works, I just played with the numbers until I saw a pattern.